cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -216,6 +218,13 @@ public void launch(RuntimeJobInfo runtimeJobInfo) throws Exception {
       }
       DataprocUtils.emitMetric(provisionerContext, region,
                                "provisioner.submitJob.response.count");
+    } catch (ResourceExhaustedException ex) {
+      // delete all uploaded gcs files in case of exception
+      DataprocUtils.deleteGCSPath(getStorageClient(), bucket, runRootPath);
```

Comment:
These two lines are duplicated. You can move them to finally, or catch multiple exceptions. You can even do something like if (ex instanceof ResourceExhaustedException) {throw new TooMany....();}

Reply:
The gcs cleanup and metric emission is to be done only when job submission fails. The `finally` block would be executed in success case too, so add an `instanceof` check to prevent duplication.